pmd-ruleset/src/main/resources/pmd-ruleset/ruleset.xml

```diff
@@ -29,4 +29,7 @@
   <rule ref="category/java/codestyle.xml/ControlStatementBraces" />
   
   <rule ref="category/java/bestpractices.xml/MissingOverride" />
+  
+  <rule ref="category/java/errorprone.xml/DoNotThrowExceptionInFinally" />
```

Comment:
so this would fail the build for code like this from [the PMD page](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#donotthrowexceptioninfinally):
```java
public class Foo {
    public void bar() {
        try {
            // Here do some stuff
        } catch( Exception e) {
            // Handling the issue
        } finally {
            // is this really a good idea ?
            throw new Exception();
        }
    }
}
```

but would it fail the build for code like this?
```java
public void method() {
  Resource resource = openSomeResource();
  try {
      // Here do some stuff
  } finally {
      closeOrThrow(resource);
  }
}

public void closeOrThrow(Resource resource) throws Exception {
  if (resource.isOpen()) {
    throw new Exception("problem");
  }
  else {
    resource.close();
  }
}
```

Reply:
> but would it fail the build for code like this?

It won't